components/profile/unclaimed-hypercert-claim-button.tsx

```diff
@@ -4,20 +4,44 @@ import { AllowListRecord } from "@/allowlists/getAllowListRecordsForAddressByCla
 import { Button } from "../ui/button";
 import { useHypercertClient } from "@/hooks/use-hypercert-client";
 import { waitForTransactionReceipt } from "viem/actions";
-import { useAccount, useWalletClient } from "wagmi";
+import { useAccount, useSwitchChain, useWalletClient } from "wagmi";
 import { useRouter } from "next/navigation";
+import { Row } from "@tanstack/react-table";
+import { useStepProcessDialogContext } from "../global/step-process-dialog";
+import { createExtraContent } from "../global/extra-content";
+import { revalidatePathServerAction } from "@/app/actions/revalidatePathServerAction";
+import { useState } from "react";
+
+interface UnclaimedHypercertClaimButtonProps {
+  allowListRecord: Row<AllowListRecord>;
+}
 
 export default function UnclaimedHypercertClaimButton({
   allowListRecord,
-}: {
-  allowListRecord: AllowListRecord;
-}) {
+}: UnclaimedHypercertClaimButtonProps) {
   const { client } = useHypercertClient();
   const { data: walletClient } = useWalletClient();
-  const { address } = useAccount();
+  const account = useAccount();
   const { refresh } = useRouter();
+  const [isLoading, setIsLoading] = useState(false);
+  const { setDialogStep, setSteps, setOpen, setTitle, setExtraContent } =
+    useStepProcessDialogContext();
+  const { switchChain } = useSwitchChain();
+  const selectedHypercert = allowListRecord.original;
+  const hypercertChainId = selectedHypercert?.hypercert_id?.split("-")[0];
 
   const claimHypercert = async () => {
+    setIsLoading(true);
+    setOpen(true);
+    setSteps([
+      { id: "preparing", description: "Preparing to claim hypercert..." },
+      { id: "claiming", description: "Claiming hypercert on-chain..." },
+      { id: "confirming", description: "Waiting for on-chain confirmation" },
+      { id: "route", description: "Creating your new hypercert's link..." },
+      { id: "done", description: "Claiming complete!" },
+    ]);
+
+    setTitle("Claim Hypercert from Allowlist");
```

Comment:
It's actually a fraction, because you can get multiple entries to claim a fraction from a single hypercert